File: clients/node/sdk/mastodonClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MastodonClient = void 0;
const apiRequestBuilder_1 = require("./api/apiRequestBuilder");
const oauthRequestBuilder_1 = require("./oauth/oauthRequestBuilder");
const kiota_abstractions_1 = require("@microsoft/kiota-abstractions");
const kiota_serialization_json_1 = require("@microsoft/kiota-serialization-json");
const kiota_serialization_text_1 = require("@microsoft/kiota-serialization-text");
/** The main entry point of the SDK, exposes the configuration and the fluent API. */
class MastodonClient {
    /** The api property */
    get api() {
        return new apiRequestBuilder_1.ApiRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /** The oauth property */
    get oauth() {
        return new oauthRequestBuilder_1.OauthRequestBuilder(this.pathParameters, this.requestAdapter);
    }
    /**
     * Instantiates a new MastodonClient and sets the default values.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    constructor(requestAdapter) {
        if (!requestAdapter)
            throw new Error("requestAdapter cannot be undefined");
        this.pathParameters = {};
        this.urlTemplate = "{+baseurl}";
        this.requestAdapter = requestAdapter;
        (0, kiota_abstractions_1.registerDefaultSerializer)(kiota_serialization_json_1.JsonSerializationWriterFactory);
        (0, kiota_abstractions_1.registerDefaultSerializer)(kiota_serialization_text_1.TextSerializationWriterFactory);
        (0, kiota_abstractions_1.registerDefaultDeserializer)(kiota_serialization_json_1.JsonParseNodeFactory);
        (0, kiota_abstractions_1.registerDefaultDeserializer)(kiota_serialization_text_1.TextParseNodeFactory);
        if (requestAdapter.baseUrl === undefined || requestAdapter.baseUrl === "") {
            requestAdapter.baseUrl = "https://mastodon.example";
        }
    }
    ;
}
exports.MastodonClient = MastodonClient;

File: clients/csharp/sdk/MastodonClient.cs
using MastodonClientLib.Api;
using MastodonClientLib.Oauth;
using Microsoft.Kiota.Abstractions.Extensions;
using Microsoft.Kiota.Abstractions;
using Microsoft.Kiota.Serialization.Form;
using Microsoft.Kiota.Serialization.Json;
using Microsoft.Kiota.Serialization.Text;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using System;
namespace MastodonClientLib {
    /// <summary>
    /// The main entry point of the SDK, exposes the configuration and the fluent API.
    /// </summary>
    public class MastodonClient : BaseRequestBuilder {
        /// <summary>The api property</summary>
        public ApiRequestBuilder Api { get =>
            new ApiRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The oauth property</summary>
        public OauthRequestBuilder Oauth { get =>
            new OauthRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>
        /// Instantiates a new MastodonClient and sets the default values.
        /// </summary>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public MastodonClient(IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}", new Dictionary<string, object>()) {
            ApiClientBuilder.RegisterDefaultSerializer<JsonSerializationWriterFactory>();
            ApiClientBuilder.RegisterDefaultSerializer<TextSerializationWriterFactory>();
            ApiClientBuilder.RegisterDefaultSerializer<FormSerializationWriterFactory>();
            ApiClientBuilder.RegisterDefaultDeserializer<JsonParseNodeFactory>();
            ApiClientBuilder.RegisterDefaultDeserializer<TextParseNodeFactory>();
            ApiClientBuilder.RegisterDefaultDeserializer<FormParseNodeFactory>();
            if (string.IsNullOrEmpty(RequestAdapter.BaseUrl)) {
                RequestAdapter.BaseUrl = "https://mastodon.example";
            }
            PathParameters.TryAdd("baseurl", RequestAdapter.BaseUrl);
        }
    }
}
